src/pages/QuizUi/QuizContent.jsx

```diff
@@ -0,0 +1,172 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import useQuiz from '../../quiz/useQuiz';
+import { useParams } from 'react-router-dom';
+import { Button, Container, Typography } from '@mui/material';
+import QuestionContent from './QuestionContent';
+import Loading from 'components/Loading';
+import { QUIZZES } from '../../App';
+import customColors from 'assets/styles';
+import { containerStyles, titleStyles } from '../Main';
+
+export default function QuizContent() {
+  const { quizId } = useParams();
+  const { quizzes } = useQuiz();
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [quizFinished, setQuizFinished] = useState(false);
+  const [answersCorrectness, setAnswersCorrectness] = useState([]);
+  const [userPerformance, setUserPerformance] = useState({
+    correctCount: 0,
+    totalCount: 0,
+  });
+
+  // Rest of your component logic...
+  const quiz = quizzes.find((q) => q.id === quizId);
+  const questions = quiz?.questions || [];
+  const quizTitle = quiz?.title || '';
+
+  useEffect(() => {
+    if (quizFinished) {
+      const correctCount = answersCorrectness.filter(
+        (answer) => answer && answer.isCorrect
+      ).length;
+      const totalCount = questions.length;
+      setUserPerformance({ correctCount, totalCount });
+    }
+  }, [quizFinished, answersCorrectness, questions.length]);
+
+  const handleAnswerSelected = (answer, resetCallback) => {
```

Comment:
Great use of naming a callback to aid the reader!

Reply:
Thanks Eli!